### PR TITLE
Fix Flaky test IndicesRequestCacheIT.testStaleKeysCleanupWithMultipleIndices

### PR DESCRIPTION
**Pull request by kiranprakash154**
### Description
Fix Flaky test `IndicesRequestCacheIT.testStaleKeysCleanupWithMultipleIndices.` failing with `java.lang.AssertionError: expected:<-665> but was:<665>`

In the below code, memorySizeOfLatestEntryForIndex1 would be negative (-665) if the cacheCleaner ran before the execution of that line resulting in `getRequestCacheStats(client, index1).getMemorySizeInBytes()` to be smaller than `finalMemorySizeForIndex1`

```java 
long memorySizeOfLatestEntryForIndex1 = getRequestCacheStats(client, index1).getMemorySizeInBytes() - finalMemorySizeForIndex1;
.
.
.
assertEquals(memorySizeOfLatestEntryForIndex1, getRequestCacheStats(client, index1).getMemorySizeInBytes())
```

#### What is the Fix ?
The fix is to not depend on the exact size of the `memorySizeOfLatestEntryForIndex1` and instead make sure the current request cache stats size is less than `memorySizeForIndex1With2Entries` and `> 0`
By this, we avoid the race condition of cacheCleaner being invoked before the calculation of `memorySizeOfLatestEntryForIndex1`

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/13437

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kiranprakash154** [comment]
> ❌ Gradle check result for [d81d42a](https://github.com/opensearch-project/OpenSearch/commit/d81d42ad43f0d1e87ab08357641a015e540fa8a1): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/37981/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md?rgh-link-date=2024-04-29T20%3A40%3A41Z#flaky-tests) unrelated to your change?

Tests with failures:
 - org.opensearch.snapshots.CloneSnapshotIT.testShallowCloneNameAvailability - https://github.com/opensearch-project/OpenSearch/issues/10155
 - org.opensearch.remotemigration.RemoteReplicaRecoveryIT.testReplicaRecovery - https://github.com/opensearch-project/OpenSearch/pull/9682 & https://github.com/opensearch-project/OpenSearch/issues/8919

**Review by dblock — APPROVED** [review]

File: server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheIT.java [hunk 1]
```diff
@@ -1194,37 +1202,40 @@ public void testStaleKeysCleanupWithMultipleIndices() throws Exception {
...
+        // sleep until cache cleaner would have cleaned up the stale key from index2
         assertBusy(() -> {
-            // cache cleaner should have cleaned up the stale key from index 2
+            // cache cleaner should have cleaned up the stale key from index2 and hence cache should be empty
             assertEquals(0, getRequestCacheStats(client, index2).getMemorySizeInBytes());
-            // cache cleaner should have only cleaned up the stale entities
-            assertEquals(memorySizeOfLatestEntryForIndex1, getRequestCacheStats(client, index1).getMemorySizeInBytes());
+            // cache cleaner should have only cleaned up the stale entities for index1
+            long currentMemorySizeInBytesForIndex1 = getRequestCacheStats(client, index1).getMemorySizeInBytes();
+            assertTrue(currentMemorySizeInBytesForIndex1 < memorySizeForIndex1With2Entries);
```
Comment (sgup432) [inline server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheIT.java:1236]: The last `createCacheEntry(client, index1, "hello");` would be equal to  `memorySizeForIndex1With1Entries` as it is the same entry?
 If yes, can we use exact match?
```
assertTrue(memorySizeForIndex1With1Entries, currentMemorySizeInBytesForIndex1);
```

Comment (kiranprakash154) [reply server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheIT.java]: Makes sense, i guess u meant 
```
assertEquals(memorySizeForIndex1With1Entries, currentMemorySizeInBytesForIndex1);
```

**Review by sgup432 — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Comment by kiranprakash154** [comment]
> ❕ Gradle check result for [5b6926e](https://github.com/opensearch-project/OpenSearch/commit/5b6926e300944b6b7b15f18b842ca8b5a923283b): [UNSTABLE](https://build.ci.opensearch.org/job/gradle-check/37996/)
> 
> * **TEST FAILURES:**
> 
> ```
>       1 org.opensearch.index.IndexServiceTests.testAsyncTranslogTrimTaskOnClosedIndex
> ```
> 
> Please review all [flaky tests](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md?rgh-link-date=2024-04-29T22%3A44%3A49Z#flaky-tests) that succeeded after retry and create an issue if one does not already exist to track the flaky failure.

Known Flaky https://github.com/opensearch-project/OpenSearch/issues/11547

**Comment by kiranprakash154** [comment]
@dblock can u merge this ?

**Comment by dblock** [comment]
Happened again in https://github.com/opensearch-project/OpenSearch/pull/13655.

**Pull request merged**